### DISCUSSION TITLE
Formatting NativeSQL query result

### DISCUSSION THREAD

**Discussion by TristanPouliquen** [symfony/symfony#42845] [Q&A]
Hello, I have a small Native Query question.

I set up my query like this:
$rsm = new ResultSetMapping();
        $rsm->addEntityResult(PspAggregate::class, 'pA');
        $rsm->addFieldResult('pA', 'id', 'id');

        $query = $this->entityManager->createNativeQuery(
            "SELECT pA.id FROM psp_aggregate pA " .
            "LEFT JOIN payout p ON p.psp_aggregate_id = pA.id " .
            "WHERE p.id IS NULL AND pA.action = ?"
            ,
            $rsm
        );
        $query->setParameter(1, PspAggregateInterface::ACTION_PAYOUT);

        return $query->getScalarResult();
This returns me an array of array of my results, under the form [['pA_id' => ...]] .
Does someone see a way for me to easily & cleanly (ie: using Doctrine) have my results under the form [['id' => ...]] ?
I could use PHP to transform my array accordingly but I was wondering if a Doctrine way existed to keep on learning 😁 

Thanks in advance if someone can help me!

**Comment by Nek-** [comment]
Why not using directly the dbal?

```
$this->entityManager->connection->query('SELECT ...');
```

Doc here: https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/data-retrieval-and-manipulation.html#data-retrieval

Just do you know, there's also a method `getArrayResult()` on `Query` objects.

**Reply by TristanPouliquen** [reply]
I wanted to test a bit more the NativeQuery & the ResultSetMapping to discover a bit more about these tools.

But I'll fall back to other solutions if I do not find a way to make it work, it just seems less "Symfony/Doctrine" 😅

**Reply by Nek-** [reply]
How Doctrine dbal can be less "doctrine" 😜 .

Using the ResultSetMapper (RSM) to make your request is exactly what Doctrine does internally. This is the process of binding the result of a query to objects. Since you expect an array of items and not an array of objects, using the result set mapper makes no sense.

But I missed the point, your issue is that you want `[['id' => ...]]` instead of `[['pA_id' => ...]]`... Did you try in your SQL request to make this? `SELECT pA.id as id` I have no idea if it works, but maybe...
